### ISSUE TITLE
Error: You must specify a valid date and/or time.

### ISSUE THREAD

**Issue by shontejnr** [phw198/OutlookGoogleCalendarSync#217]
Please see below for a section of the log. This prompts with an error meaning I have to manual sync. Sync appears fine on the whole so need some help to overcome this please. Thanks.
```
2017-05-25 14:34:20,348 INFO 12 OutlookGoogleCalendarSync.MainForm [0] - WARNING: Appointment update failed.
You must specify a valid date and/or time. Check your entries in this dialog box to verify they represent a valid date and/or time.
2017-05-25 14:34:20,348 ERROR 12 OutlookGoogleCalendarSync.OutlookCalendar [0] -    at Microsoft.Office.Interop.Outlook.RecurrencePattern.set_PatternEndDate(DateTime PatternEndDate)
   at OutlookGoogleCalendarSync.Recurrence.buildOutlookPattern(Event ev, AppointmentItem ai, RecurrencePattern& oPattern)
   at OutlookGoogleCalendarSync.Recurrence.CompareOutlookPattern(Event ev, RecurrencePattern& aiOpattern, SyncDirection syncDirection, StringBuilder sb, Int32& itemModified)
   at OutlookGoogleCalendarSync.OutlookCalendar.UpdateCalendarEntry(AppointmentItem& ai, Event ev, Int32& itemModified, Boolean forceCompare)
   at OutlookGoogleCalendarSync.OutlookCalendar.UpdateCalendarEntries(Dictionary`2 entriesToBeCompared, Int32& entriesUpdated)
2017-05-25 14:34:22,100 INFO 12 OutlookGoogleCalendarSync.MainForm [0] - 0 entries updated.
2017-05-25 14:34:22,217 INFO 12 OutlookGoogleCalendarSync.MainForm [0] - --------------------------------------------------
```

**Comment by phw198** [comment]
I can't tell from that limited information. The log doesn't show the section before which should include a debug line `Converting RRULE:....` - I at least need that line to troubleshoot this. I don't even know the version of OGCS you're running...

**Comment by shontejnr** [comment]
Sorry, tried to add the text file but says I cant. Here is more info...
```
2017-05-25 15:36:43,343 WARN 7 OutlookGoogleCalendarSync.OutlookCalendar [0] - Could not find Google calendar ID against Outlook appointment item.
2017-05-25 15:36:43,348 WARN 7 OutlookGoogleCalendarSync.OutlookCalendar [0] - Could not find Google calendar ID against Outlook appointment item.
2017-05-25 15:36:43,352 WARN 7 OutlookGoogleCalendarSync.OutlookCalendar [0] - Could not find Google calendar ID against Outlook appointment item.
2017-05-25 15:36:43,357 WARN 7 OutlookGoogleCalendarSync.OutlookCalendar [0] - Could not find Google calendar ID against Outlook appointment item.
2017-05-25 15:36:43,362 WARN 7 OutlookGoogleCalendarSync.OutlookCalendar [0] - Could not find Google calendar ID against Outlook appointment item.
2017-05-25 15:36:43,366 WARN 7 OutlookGoogleCalendarSync.OutlookCalendar [0] - Could not find Google calendar ID against Outlook appointment item.
2017-05-25 15:36:43,367 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - 0 Outlook calendar entries to be deleted.
2017-05-25 15:36:43,367 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - 0 Outlook calendar entries to be created.
2017-05-25 15:36:43,367 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - --------------------------------------------------
2017-05-25 15:36:43,367 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - Comparing 793 existing Outlook calendar entries...
2017-05-25 15:36:45,459 DEBUG 7 OutlookGoogleCalendarSync.OutlookCalendar [0] - Processing recurring master appointment.
2017-05-25 15:36:45,459 DEBUG 7 OutlookGoogleCalendarSync.OutlookCalendar [0] - Processing >> 28/05/2010 (R) => "Ryan Prins's Birthday"
2017-05-25 15:36:45,497 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - Converting RRULE:FREQ=YEARLY;UNTIL=45001231T000000Z;INTERVAL=1
2017-05-25 15:36:45,517 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - WARNING: Appointment update failed.
You must specify a valid date and/or time. Check your entries in this dialog box to verify they represent a valid date and/or time.
2017-05-25 15:36:45,517 ERROR 7 OutlookGoogleCalendarSync.OutlookCalendar [0] -    at Microsoft.Office.Interop.Outlook.RecurrencePattern.set_PatternEndDate(DateTime PatternEndDate)
   at OutlookGoogleCalendarSync.Recurrence.buildOutlookPattern(Event ev, AppointmentItem ai, RecurrencePattern& oPattern)
   at OutlookGoogleCalendarSync.Recurrence.CompareOutlookPattern(Event ev, RecurrencePattern& aiOpattern, SyncDirection syncDirection, StringBuilder sb, Int32& itemModified)
   at OutlookGoogleCalendarSync.OutlookCalendar.UpdateCalendarEntry(AppointmentItem& ai, Event ev, Int32& itemModified, Boolean forceCompare)
   at OutlookGoogleCalendarSync.OutlookCalendar.UpdateCalendarEntries(Dictionary`2 entriesToBeCompared, Int32& entriesUpdated)
2017-05-25 15:36:47,448 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - User chose not to continue sync.
2017-05-25 15:36:53,580 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - Reading Outlook Calendar Entries...
2017-05-25 15:36:55,475 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - 811 Outlook calendar entries found.
2017-05-25 15:36:55,475 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - --------------------------------------------------
2017-05-25 15:36:55,475 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - Reading Google Calendar Entries...
2017-05-25 15:36:55,475 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Retrieving all events from Google: 25/05/2016 -> 26/05/2018
2017-05-25 15:36:56,072 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 1 received.
2017-05-25 15:36:56,589 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 2 received.
2017-05-25 15:36:57,505 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 3 received.
2017-05-25 15:36:57,921 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 4 received.
2017-05-25 15:36:58,422 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 5 received.
2017-05-25 15:36:58,891 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 6 received.
2017-05-25 15:36:59,371 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 7 received.
2017-05-25 15:36:59,740 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 8 received.
2017-05-25 15:36:59,740 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - 804 Google calendar entries found.
2017-05-25 15:36:59,740 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - Identifying exceptions in recurring Google events.
2017-05-25 15:36:59,740 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - Found 11 exceptions.
2017-05-25 15:36:59,740 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - 11 are exceptions to recurring events.
2017-05-25 15:36:59,740 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - --------------------------------------------------
2017-05-25 15:36:59,740 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - Total inc. recurring items spanning sync date range...
2017-05-25 15:37:06,116 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Retrieving specific Event with ID kh1mf249qlat9g9m19qaddlduo
2017-05-25 15:37:09,587 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Retrieving specific Event with ID 290u565k7ul6c5g8rtk0dung7o
2017-05-25 15:37:16,849 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Retrieving specific Event with ID apfn86d1icufcilavqsm1sp2us
2017-05-25 15:37:17,293 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - Outlook 793, Google 793
2017-05-25 15:37:17,293 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - --------------------------------------------------
2017-05-25 15:37:17,293 DEBUG 7 OutlookGoogleCalendarSync.MainForm [0] - Synchronising from Outlook to Google.
2017-05-25 15:37:17,293 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Scanning 793 Google events for orphans to reclaim...
2017-05-25 15:37:17,293 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - 0 unclaimed.
2017-05-25 15:37:17,293 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Comparing Outlook items to Google events...
2017-05-25 15:37:20,572 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - 0 Google calendar entries to be deleted.
2017-05-25 15:37:20,572 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - 0 Google calendar entries to be created.
2017-05-25 15:37:20,572 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - --------------------------------------------------
2017-05-25 15:37:20,572 INFO 7 OutlookGoogleCalendarSync.MainForm [0] - Comparing 793 existing Google calendar entries...
2017-05-25 15:37:21,173 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - 20/12/2011 13:00 (R) => "CFC"
2017-05-25 15:37:21,173 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - This is a recurring appointment with 86 exceptions that will now be iteratively compared.
2017-05-25 15:37:23,274 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - Finding Google instance for deleted Outlook exception:-
2017-05-25 15:37:23,274 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] -   Original date: 07/06/2016
2017-05-25 15:37:23,274 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - Matching Google Event recurrence found.
2017-05-25 15:37:23,274 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - It is deleted in Google, so cannot compare items.
2017-05-25 15:37:23,327 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - Finding Google instance for Outlook exception:-
2017-05-25 15:37:23,327 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] -   Original date: 28/06/2016
2017-05-25 15:37:23,327 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] -   Current  date: 29/06/2016
2017-05-25 15:37:23,343 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - Google exception event is not cached. Retrieving all recurring instances...
2017-05-25 15:37:23,343 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Retrieving all recurring event instances from Google.
2017-05-25 15:37:23,858 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 1 received.
2017-05-25 15:37:24,344 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 2 received.
2017-05-25 15:37:24,760 DEBUG 7 OutlookGoogleCalendarSync.GoogleCalendar [0] - Page 3 received.
2017-05-25 15:37:24,776 DEBUG 7 OutlookGoogleCalendarSync.Recurrence [0] - Matching Google Event recurrence found.
```

**Comment by phw198** [comment]
For future reference, [guidance for providing your logfile](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-problems#posting-your-ogcalsynclog-file).

So "Ryan Prins's Birthday" Google appointment has an end date of 31 Dec 4500 - Outlook can't handle that. Do you know how this Google appointment was created? The Google webapp doesn't allow it...

**Comment by phw198** [comment]
:package: This should now be resolved with [Hotfix-v2.4.0.4.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1029788/Hotfix-v2.4.0.4.zip) [[see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)]

Please let me know how it goes.

**Comment by shontejnr** [comment]
That appears to be working great now, thank you so much for the prompt response.
